channels/channels_fsm.go

```diff
@@ -103,6 +104,24 @@ var ChannelEvents = fsm.Events{
 			chst.AddLog("")
 			return nil
 		}),
+
+	fsm.Event(datatransfer.CIDMissing).FromMany(transferringStates...).ToJustRecord().
+		Action(func(chst *internal.ChannelState, missing cid.Cid) error {
+			// TODO: find a more efficient way to do this
```

Comment:
Would it make sense to store the missing cids as a map[cid.Cid]struct{} instead?

Reply:
Ok 👍